src/lib/requests/common.js

```diff
@@ -65,8 +68,36 @@ const throwOrJson = async (res) => {
     return res.data;
 };
 
+const defineAgent = (url) =>  {
+    const httpModule = url.startsWith('https') ? https : http;
+    // make sure we keep alive connections to the backend
+    return new httpModule.Agent({
+        keepAlive: true,
+    });
+}
```

Comment:
I don't thinks this is needed since we'll be using Istio sidecar injection which give us free mTLS behind the scenes. Ref: https://istio.io/latest/docs/concepts/security/#mutual-tls-authentication

Reply:
Sorry, i didn't completely get, what is not needed - using Agent with keepAlve option or what?

I've tried not to break existing logic, so I just moved this code in a separate file, coz it was used in a couple of places:

- https://github.com/pm4ml/mcm-client/pull/6/files#diff-166152a9b4beec6af970401bb604007a91f007894bb2e04027b29a7411f89805L17-L20
- https://github.com/pm4ml/mcm-client/pull/6/files#diff-3d9208142e32ad039de937f4dff0e687f503a4b852521f25d2c91b81067e6568L28-L31

And i've added support for httpS sheme, coz now, PM first will need to do a direct call to Keycloak on hub through Internet to get JWT tokens (here's how Keycloack url looks on dev k8s cluster: https://keycloak.dev.labsk8s58.mojaloop.live/realms/dfsps/protocol/openid-connect/token)
Here's a diagram:
https://i.imgur.com/liS4IW0.png

Reply:
Ah, totally makes sense now. Thanks for the clarification.